From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Thu, 4 Sep 2025 22:40:46 +0200
Subject: fix(udev-rules): exclude udev rules from snapd

There are no snaps in the initrd and these udev rules generated by snapd
can cause warning messages. So just exclude udev rules in `/etc` that
have `70-snap` in the name.

Forwarded: https://github.com/dracut-ng/dracut-ng/pull/1620
---
 modules.d/74udev-rules/module-setup.sh | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/modules.d/74udev-rules/module-setup.sh b/modules.d/74udev-rules/module-setup.sh
index 35e4399..2772f5b 100755
--- a/modules.d/74udev-rules/module-setup.sh
+++ b/modules.d/74udev-rules/module-setup.sh
@@ -102,9 +102,19 @@ install() {
     # Install the hosts local user configurations if enabled.
     if [[ ${hostonly-} ]]; then
         inst_dir "$udevconfdir"
+
+        declare -a udevrules=()
+        for rule in "$udevrulesconfdir"/*.rules; do
+            [ -e "$rule" ] || continue
+            if [[ $rule =~ "$udevrulesconfdir"/70-snap.*.rules ]]; then
+                continue
+            fi
+            udevrules+=("$rule")
+        done
+
         inst_multiple -H -o \
             "$udevconfdir"/udev.conf \
             "$udevconfdir/udev.conf.d/*.conf" \
-            "$udevrulesconfdir/*.rules"
+            "${udevrules[@]}"
     fi
 }
